packages/angular_devkit/core/node/host.ts

```diff
@@ -316,6 +317,9 @@ export class NodeJsSyncHost implements virtualFs.Host<fs.Stats> {
       // TODO: remove this try+catch when issue https://github.com/ReactiveX/rxjs/issues/3740 is
       // fixed.
       try {
+        if (!fs.existsSync(getSystemPath(path.parse(to).dir as Path))) {
```

Comment:
The castings here are not correct The `to` and `from` here are devkit paths are they are not interchangeable with FS paths, so you need to get the system path first.

So the logic here should look more like;
```ts
try {
  const toSystemPath = getSystemPath(to);
  if (!fs.existsSync(path.dirname(toSystemPath)) {
      fs.mkdirSync(path.dirname(toSystemPath), { recursive: true });
  }

  fs.renameSync(getSystemPath(from), toSystemPath);
  obs.next();
  obs.complete();
} catch (err) {
  obs.error(err);
}
```